isodatetime/data.py

```diff
@@ -146,175 +148,6 @@ def __str__(self):
         return format_string.format(*format_args)
 
 
-class TimeRecurrence(object):
```

Comment:
Not deleted, only moved after `TimePoint` as it needs to be defined before it, so we can reference an argument to be typed as `TimePoint`.